server/app.js

```diff
@@ -1,246 +1,275 @@
+            payload = payload.split(":");
+            debug_log(payload);            
+            
+            switch(packetType)
+            {
+                case 0: socket.broadcast.emit('infoDiag', payload); break;
+                case 1: socket.broadcast.emit('errorDiag', payload); break;
+                case 2: socket.broadcast.emit('statusDiag', payload); break;
+                case 3: socket.broadcast.emit('log', payload); break;
+            }
+        });
+    });
+
+    server.listen(serverListenPort, serverListenAddress); 
+    ///////////////////////////////////////////////////////////////// 
+
+
+    var connected = false;
+    socket.on('stream',function(image){
+        socket.broadcast.emit('stream', image);
+    });
+    socket.on('err', function(err){
+            // handle the error safely
+            debug_log(err)
+        });
+    socket.on('event',function(data){
+        
+        if(data.message == 20) {
+            clearInterval(repeat);
+            data.message = 14;
+        }
+
+        clientConnect(data.message, 0xA);
+    });
+    socket.on('dirEvent',function(data){
+        clientConnect(data.message, 0x32);
+    });    
+    socket.on('dutyEvent',function(data){
+        clientConnect(12, data.message);
+    });
+    socket.on('exit', function(ignore) { 
+        socket.disconnect(true);
+        debug_log("Server disconnecting.");
+        process.exit(); 
+    });
+});
+
+io.on('err', function(err){
+    // handle the error safely
+    debug_log(err)
+})
+
+http.listen(port,function(){
+    log.info('Server listening through port %s', port);
+});
+
+http.on('err', function(err){
+   // handle the error safely
+    debug_log(err)
+});
+
+process.on('uncaughtException', function(err) {
+  debug_log('Caught exception: ' + err);
+});
+
+
+const state = {"NAVIGATING_SIDEWALK":0, "LOOKING_FOR_PATH":1 ,"FOUND_PATH_TO_HOUSE":2, "NAVIGATING_PATH":3 ,"REACHED_DOOR":4, "DELIVER_PACKAGE":5 }
+var current_state;
+var repeat;
+var state_repeat;
+call_number_recognition();
+call_video_stream();
+//init_state_machine();
+
+function init_state_machine() {
+    // Initial state Navigating Sidewalk
+
+    clientConnect(14, 0x4);
+
+    repeat = setInterval(navigating_sidewalk, 500);
+}
+
+function restart_state_machine(){
+    stop_state_machine();
+    init_state_machine();
+}
+
+function stop_state_machine(){
+    clearInterval(repeat);
+    clearInterval(state_repeat);
+}
+
+function call_video_stream() {
+    debug_log("Calling video stream script.\n");
+    var pyshell = new PythonShell('../sidewalk/python/video_stream.py')
+}
+
+function navigating_sidewalk(){
+    //Read
+    current_state = state["NAVIGATING_SIDEWALK"];
+
+    debug_log("navigating sidewalk");
+
+    var file = 'front.json';
+    jsonfile.readFile(file, function(err, obj) {
+        //console.log(obj.angle);
+
+        if(obj.angle >= 85 && obj.angle <= 95 ){
+            clientConnectSendState(0,0x1E,0);       
```

Comment:
I think we can make this stuff cleaner.
Documented in #20 .
But to merge this asap I would say do it in a separate PR.